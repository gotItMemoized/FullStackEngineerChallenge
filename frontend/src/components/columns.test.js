import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Columns from './Columns';
import 'jest-dom/extend-expect';

afterEach(cleanup);

it('renders without crashing', () => {
  const { container } = render(<Columns />);
  expect(container).toMatchSnapshot();
});

it('renders with classname', () => {
  const { container } = render(<Columns className="test" />);
  expect(container).toMatchSnapshot();
});

it('renders children', () => {
  const { container } = render(
    <Columns>
      <div id="something">hello</div>
    </Columns>,
  );
  expect(container).toMatchSnapshot();
  const innerDiv = document.getElementById('something');
  expect(innerDiv).toMatchSnapshot();
  expect(innerDiv).toHaveTextContent('hello');
});
