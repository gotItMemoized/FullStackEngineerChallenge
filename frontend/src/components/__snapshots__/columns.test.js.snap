// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders children 1`] = `
<div>
  <div
    class="columns "
  >
    <div
      id="something"
    >
      hello
    </div>
  </div>
</div>
`;

exports[`renders children 2`] = `
<div
  id="something"
>
  hello
</div>
`;

exports[`renders with classname 1`] = `
<div>
  <div
    class="columns test"
  />
</div>
`;

exports[`renders without crashing 1`] = `
<div>
  <div
    class="columns "
  />
</div>
`;
